src/span.js

```diff
@@ -0,0 +1,165 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import * as constants from './constants.js';
+import * as thrift from './thrift.js';
+import Utils from './util.js';
+import SpanContext from './span_context.js';
+
+export default class Span {
+    _tracer: any;
+    _name: string;
+    _spanContext: SpanContext;
+    _start: number;
+    _duration: number;
+    _firstInProcess: boolean;
+    _isClient: boolean;
+    _peer: Endpoint;
+    _annotations: Array<Annotation>;
+    _binaryAnnotations: Array<BinaryAnnotation>;
+
+    constructor(tracer: any,
+                name: string,
+                spanContext: SpanContext,
+                start: number = Utils.getTimestampMicros(),
+                firstInProcess: boolean = false,
+                annotations: Array<Annotation> = [],
+                binaryAnnotations: Array<BinaryAnnotation> = []
+    ) {
+        this._tracer = tracer;
+        this._name = name;
+        this._spanContext = spanContext;
+        this._start = start;
+        this._firstInProcess = firstInProcess;
+        this._duration;
+        this._isClient;
+        this._peer;
+
+        // logs
+        this._annotations = annotations;
+
+        // tags
+        this._binaryAnnotations = binaryAnnotations;
+    }
+
+    /**
+     * Returns the span context that represents this span.
+     *
+     * @return {SpanContext} - Returns this span's span context.
+     **/
+    context(): SpanContext {
+        return this._spanContext;
+    }
+
+    /**
+     * Returns the tracer associated with this span.
+        this._duration;
+     * @return {Tracer} - returns the tracer associated witht this span.
+     **/
+    tracer(): any {
+        return this._tracer;
+    }
+
+    /**
+     * Sets the operation name on this given span.
+     *
+     * @param {string} name - The name to use for setting a span's operation name.
+     **/
+    setOperationName(name: string): void {
+        this._name = name;
+    }
+
+    /**
+     * Finishes a span which has the effect of reporting it and
+     * setting the finishTime on the span.
+     *
+     * @param {number} finishTime - The time on which this span finished.
+     **/
+    finish(finishTime: number): void {
+        if (this._duration) {
```

Comment:
this should be something like `this._duration === undefined` or whatever the syntax is. Afaik we're only getting millisecond precision timestamps in JS, so it's quite possible for a duration of a finished span to be 0 if it was faster that 1ms.